File: acm/accepted/102.pas
program p102;
{
	@JUDGE_ID:  20044WC 102 PASCAL

	Oh damnit....... This was easier than I had thought......
}
const
	p : array[1..6] of string[3] = ('BCG', 'BGC', 'CBG', 'CGB', 'GBC', 'GCB');
var
	i,j,k : longint;
	m : longint;
	mp : string[3];
	t : longint;
	a : array[1..3,'B'..'G'] of longint;
begin
	while not eof do begin
		for i := 1 to 3 do begin
			read(a[i,'B']);
			read(a[i,'G']);
			read(a[i,'C']);
		end;
		readln;
		
		m := -1;
		for i := 1 to 6 do begin {try all configs}
			t := 0;
			for j := 1 to 3 do begin {3 boxes}
				if p[i,j] <> 'B' then t := t + a[j, 'B'];
				if p[i,j] <> 'C' then t := t + a[j, 'C'];
				if p[i,j] <> 'G' then t := t + a[j, 'G'];
			end;
			if (m = -1) or ( t < m ) then begin
				m := t;
				mp := p[i];
			end;
		end;
		writeln(mp, ' ', m);
	end;
end.

File: lib/cpp/102.cpp
#include <iostream>

int main() {
    int B[3];
    int G[3];
    int C[3];
    while (std::cin >> B[0] >> G[0] >> C[0] >> B[1] >> G[1] >> C[1] >> B[2] >> G[2] >> C[2]) {
        // BCG,BGC,CBG,CGB,GBC,GCB
        size_t total = B[0]+B[1]+B[2]+G[0]+G[1]+G[2]+C[0]+C[1]+C[2];
        size_t e = 1ULL << 62;
        std::string cfg = "";
        size_t cur;

        // BCG
        cur = total - B[0] - C[1] - G[2];
        if (cur < e) {
            cfg = "BCG";
            e = cur;
        }

        // BGC
        cur = total - B[0] - C[2] - G[1];
        if (cur < e) {
            cfg = "BGC";
            e = cur;
        }

        // CBG
        cur = total - B[1] - C[0] - G[2];
        if (cur < e) {
            cfg = "CBG";
            e = cur;
        }

        // CGB
        cur = total - B[2] - C[0] - G[1];
        if (cur < e) {
            cfg = "CGB";
            e = cur;
        }

        // GBC
        cur = total - B[1] - C[2] - G[0];
        if (cur < e) {
            cfg = "GBC";
            e = cur;
        }

        // GCB
        cur = total - B[2] - C[1] - G[0];
        if (cur < e) {
            cfg = "GCB";
            e = cur;
        }

        std::cout << cfg << " " << e << std::endl;
    }
}
